flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/sql/TableWithSQLITCase.scala

```diff
@@ -113,4 +116,31 @@ class TableWithSQLITCase(
     val results = result.toDataSet[Row].collect()
     TestBaseUtils.compareResultAsText(results.asJava, expected)
   }
+
+  /** test insert into **/
+  @Test
+  def testInsertIntoTable(): Unit = {
+    val tmpFile = File.createTempFile("flink-sql-table-sink-test1", ".tmp")
+    tmpFile.deleteOnExit()
+    val path = tmpFile.toURI.toString
+
+    val env = ExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env)
+
+    val t = CollectionDataSets.getSmall3TupleDataSet(env).toTable(tEnv).as('a, 'b, 'c)
+    tEnv.registerTable("sourceTable", t)
+
+    val fieldTypes = tEnv.scan("sourceTable").getSchema.getTypes
+    val fieldNames = Seq("d", "e", "f").toArray
+    val sink = new CsvSQLTableSink(path, fieldTypes, fieldNames, ",")
```

Comment:
We could also implement a `TableSink` which stores results in a `static List` guarded by a lock to synchronize this inserts. This would make the test faster and more reliable (going of the the filesystem is a reason for flaky tests).